packages/terra-pills/src/Pill.jsx

```diff
@@ -0,0 +1,210 @@
+  /**
+   * The label text for the pill. (Required)
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The html `id` attribute for the pill.
+   */
+  id: PropTypes.string,
+  /**
+   * A callback function to execute when the pill is removed. If specified, the pill will be removable.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * A callback function to execute when the pill is selected. If specified, the pill label will be selectable.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Design-Standards/blue) Intended to only be used to disclose a popup.
+   */
+  onSelect: PropTypes.func,
+  /**
+   * Callback to expose pill element's ref for popup placement.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to set the `aria-haspopup`
+   * attribute to `true` when using a popup via 'onSelect'. Only applies when used along with the 'onSelect' prop.
+   */
+  ariaHasPopup: PropTypes.bool,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to set the `aria-expanded` attribute to `true`
+   * when using a popup via 'onSelect' and the popup is visible. Only applies when used along with the 'onSelect' prop.
+   */
+  ariaExpanded: PropTypes.bool,
+  /**
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) Consumers need to provide the string of the ID for the
+   * html element containing the popup content, so the `aria-controls` attribute can correctly be associated to the pill.
+   * Only applies when used along with the 'onSelect' prop.
+   */
+  ariaControls: PropTypes.string,
+  /**
+   * Tooltip to display if the pill label does not have enough space to display and will show as truncated, to be used
+   * in addition to a popup. Only applies when used along with the 'onSelect' prop.
+   */
+  title: PropTypes.string,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  id: undefined,
+  onRemove: undefined,
+  onSelect: undefined,
+  refCallback: undefined,
+  ariaHasPopup: false,
+  ariaExpanded: false,
+  ariaControls: undefined,
+  title: undefined,
+};
+
+const Pill = (props) => {
+  const {
+    label,
+    id,
+    onRemove,
+    onSelect,
+    refCallback,
+    ariaHasPopup,
+    ariaExpanded,
+    ariaControls,
+    title,
+    intl,
+    ...customProps
+  } = props;
+
+  const pillRef = useRef();
+
+  useEffect(() => {
+    if (refCallback) {
+      refCallback(pillRef.current);
+    }
+  }, [refCallback]);
+
+  const handleOnClick = () => {
+    onSelect();
+  };
+
+  const handleOnRemove = () => {
+    onRemove();
+  };
+
+  const handleOnKeyDown = (event) => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'true');
+    if ((event.keyCode === KEY_RETURN || event.keyCode === KEY_SPACE) && onSelect) {
+      onSelect();
```

Comment:
You'll want to call preventDefault on the event in these cases. The demo shows the dev-site page scrolling when a disclosure Pill is selected with Space presses. I'd consider rendering a semantic `<button>` element rather than a div when the pill is selectable. It isn't easy to adequately turn a div into a button, and using a button element would prevent bugs like this.

Reply:
Will add a preventDefault(), I need to investigate how much changes is necessary(given the timeline is very narrow at this point) to render a `button` element instead of `div`.

Reply:
include preventDefault here 1319a8e97ab1b290bf12be47dce185d182b81190